app/models/concerns/geojson_format_validator.rb

```diff
@@ -3,21 +3,127 @@ def validate_each(record, attribute, value)
     if value.present?
       geojson = parse_json(value)
 
-      unless geojson?(geojson)
-        record.errors.add(attribute, :invalid)
+      unless valid_geojson?(geojson)
+        record.errors.add(
+          attribute,
+          :invalid,
+          message: I18n.t("errors.geozone.attributes.geojson.invalid")
+        )
+        return
+      end
+
+      # Validate coordinates if the GeoJSON is valid
+      if geojson["type"] == "FeatureCollection"
+        geojson["features"].each do |feature|
+          unless valid_coordinates?(feature["geometry"])
+            record.errors.add(
+              attribute,
+              :invalid_coordinates,
+              message: I18n.t("errors.geozone.attributes.geojson.invalid_coordinates")
+            )
+          end
+        end
+      elsif geojson["type"] == "Feature"
+        unless valid_coordinates?(geojson["geometry"])
+          record.errors.add(
+            attribute,
+            :invalid_coordinates,
+            message: I18n.t("errors.geozone.attributes.geojson.invalid_coordinates")
+          )
+        end
       end
```

Comment:
I'm writing some tests for this validation and it seems like we aren't validating the coordinates when we have a top-level geometry object :thinking:. Is it intentional?

Reply:
I can't really remember it is a while since I looked at this. The validation actually needs another look as it was originally done to validate the geojson when it was done using points.  If we are being correct then a geojson fragment which begins with geometry should be invalid as it isn't valid geojson as per the spec. I think I was trying to include some degree of backward compatibility to avoid breaking existing geozones which contain invalid geojson but I don't think it is totally correct

Reply:
> I can't really remember it is a while since I looked at this. 

Once again, sorry for the late review! :pray: (I wonder whether I should make that sentence part of my nickname :sweat_smile:)

> If we are being correct then a geojson fragment which begins with geometry should be invalid as it isn't valid geojson as per the spec. 

Sorry, my "top-level geometry object" comment was ambiguous. What I meant was something like:

```
{
  "type": "Point",
  "coordinates": [-73.981149, 40.747028]
}
```

If I understand the specification correctly, this is valid GeoJSON, right? So, should we check the coordinates in this case?

Reply:
By the way, just to clarify, if we need to check the coordinates in this case, I'll change the code accordingly (have already done it in a development branch in case it's needed). It's been a while since you worked on this code, so asking you to do it wouldn't be fair.